rcl/include/rcl/subscription.h

```diff
@@ -324,13 +324,15 @@ rcl_subscription_content_filter_options_init(
  * It can be NULL if there is no "%n" tokens placeholder in filter_expression.
  * \param[out] options The subscription options to be set.
  * \return `RCL_RET_OK` if set options successfully, or
+ * \return `RCL_RET_SUBSCRIPTION_INVALID` if subscription is invalid, or
  * \return `RCL_RET_INVALID_ARGUMENT` if arguments invalid, or
  * \return `RCL_RET_BAD_ALLOC` if allocating memory fails.
  */
 RCL_PUBLIC
 RCL_WARN_UNUSED
 rcl_ret_t
 rcl_subscription_content_filter_options_set(
+  const rcl_subscription_t * subscription,
```

Comment:
can this be `rcl_allocator_t`?

Reply:
[rcl_subscription_content_filter_options_init](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/include/rcl/subscription.h#L301) the `rcl_subscription_content_filter_options_t` with the rcl_subscription by using the allocator in the subscription_options,  but to use a specific `rcl_allocator_t` for [rcl_subscription_content_filter_options_set](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/include/rcl/subscription.h#L334) and [rcl_subscription_content_filter_options_fini](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/include/rcl/subscription.h#L334)?  The allocator in the rcl_subscriptions is not available for users. I think it's not convinent for user to call these two functions with a `rcl_allocator_t`. (e.x. how to fini the rcl_subscription_content_filter_options_t [here](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/test/rcl/test_subscription.cpp#L1011-L1012)?)

Reply:
> [rcl_subscription_content_filter_options_init](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/include/rcl/subscription.h#L301) the rcl_subscription_content_filter_options_t with the rcl_subscription by using the allocator in the subscription_options, but to use a specific rcl_allocator_t for [rcl_subscription_content_filter_options_set](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/include/rcl/subscription.h#L334) and [rcl_subscription_content_filter_options_fini](https://github.com/ros2/rcl/blob/05811f0e5d1cbc72cca2abb212b3f06fb7e57f71/rcl/include/rcl/subscription.h#L334)?

I meant to change it in the three.
Though your argument sounds good.